src/BMI/bmi_port.c

```diff
@@ -188,7 +188,10 @@ int bmi_start_mgr(bmi_port_mgr_t* port_mgr) {
 int bmi_port_create_mgr(bmi_port_mgr_t **port_mgr, int max_port_count) {
   bmi_port_mgr_t *port_mgr_ = malloc(sizeof(bmi_port_mgr_t));
   int exitCode;
-  if (!port_mgr) return -1;
```

Comment:
looks like this could just be moved before the `malloc` call?

Reply:
Thanks, I've updated the commit.